apps/backend/src/libs/hooks/debouncer/get-request-debouncer.hook.ts

```diff
@@ -0,0 +1,32 @@
+import { type Hook } from "../libs/types/types.js";
+import { DEBOUNCER_CONFIG } from "./libs/constants/constants.js";
+import { RequestLimitError } from "./libs/exceptions/exceptions.js";
+
+const getRequestDebouncer = (requestsLimitPerMinute: number): Hook => {
+	let debounceMap: Record<string, number> = {};
```

Comment:
let's use map for this and make it more clear what we store here
```suggestion
	const userIdToRequestNumber: Map<UserAuthResponseDto['id'], number> = new Map();
```